bson/bsoncodec/time_codec_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bsoncodec
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson/bsonoptions"
+	"go.mongodb.org/mongo-driver/bson/bsonrw/bsonrwtest"
+	"go.mongodb.org/mongo-driver/bson/bsontype"
+	"go.mongodb.org/mongo-driver/internal/testutil/assert"
+)
+
+func TestTimeCodec(t *testing.T) {
+	now := time.Now().Truncate(time.Millisecond)
+	t.Run("UseLocalTimeZone", func(t *testing.T) {
```

Comment:
We should have 3 subtests for the 3 possibilities for this option: true, false, and not set (default). We can probably do these as table driven tests.

Reply:
Done

Reply:
I'd prefer to keep the `t.Run` calls for `UseLocalTimeZone` and `DecodeFromString` and then have table-driven tests inside of those. This is completely personal preference but I think it reads better.

Reply:
Done